main/src/org/jetbrains/bsp/bazel/BazelBspServer.java

```diff
@@ -912,156 +817,104 @@ private String convertOutputToPath(String output, String prefix) {
   @Override
   public CompletableFuture<ScalacOptionsResult> buildTargetScalacOptions(
       ScalacOptionsParams scalacOptionsParams) {
-    return executeCommand(
-        () -> {
-          List<String> targets =
-              scalacOptionsParams.getTargets().stream()
-                  .map(BuildTargetIdentifier::getUri)
-                  .collect(Collectors.toList());
-          String targetsUnion = Joiner.on(" + ").join(targets);
-          Map<String, List<String>> targetsOptions = getTargetsOptions(targetsUnion, "scalacopts");
-          Either<ResponseError, ActionGraphParser> either =
-              parseActionGraph(getMnemonics(targetsUnion, Lists.newArrayList(SCALAC, JAVAC)));
-          if (either.isLeft()) return Either.forLeft(either.getLeft());
-
-          ScalacOptionsResult result =
-              new ScalacOptionsResult(
-                  targets.stream()
-                      .flatMap(
-                          target ->
-                              collectScalacOptionsResult(
-                                  either.getRight(),
-                                  targetsOptions.getOrDefault(target, new ArrayList<>()),
-                                  either.getRight().getInputsAsUri(target, getExecRoot()),
-                                  target))
-                      .collect(Collectors.toList()));
-          return Either.forRight(result);
-        });
+    return executeCommand(() -> scalaBspServer.buildTargetScalacOptions(scalacOptionsParams));
```

Comment:
Maybe this whole method should be moved to ScalaBspServer? I know that currently this is done how it is because BazelBspServer acts as a "hub" for all json APIs, but maybe it doesn't have to be that way?

Reply:
I think that there was a reason for it, we can discuss it later.

Reply:
Might I get updated on what was the reason you guys were thinking of? Since it apparently was moved either way

Reply:
@andrefmrocha 
`public class BazelBspServer implements BuildServer, ScalaBuildServer, JavaBuildServer {`
XBuildServer interfaces are responsible for JSON communication with the outside world, right? What I meant was that it would be nicer if there were separate classes, each implementing one interface

Reply:
Yeah I got that, but I thought @Abrams27 was mentioning there was something stopping us from implementing it that?

Reply:
I mentioned that there is a problem in the actual implementation with `BazelBspServer` "HUB". It can be moved to separate classes and then problem would be solved.

Reply:
I can do it in the following PR

Reply:
Sure! Sounds great, thanks for keeping me up to date